tests/bootstrap/test-bootstrap.sh

```diff
@@ -8,6 +8,9 @@ cd "$(dirname "$0")"
 earthly=${earthly-"../../build/linux/amd64/earthly"}
 earthly=$(realpath "$earthly")
 
+# docker / podman
+frontend=${frontend-"docker"}
```

Comment:
It would be helpful if `frontend` would default to docker for cases where it's not specified.

some of these tests scripts (such as this one that references `../../build/...`) are designed to be runnable via GHA and locally (at least for linux users). e.g. I can simply run `./tests/bootstrap/test-bootstrap.sh` on my local machine and this script passes.

At the very least we should test that `$frontend` exists and exit with an error message when it's missing. However, 
 I think having a default is more dev-friendly. (maybe when it's missing we could use `which` to default to docker or podman -- whichever is detected)